### PR TITLE
Extend C++ API for Map/Sequence Type Info (#3517)

### PR DESCRIPTION
**Pull request by Dudeldu**
**Description**: As discussed in the issue #3517, we added c++ functions for accessing the type information of sequences and maps. For this we introduced new methods inspired by `GetTensorTypeAndShapeInfo` to get Map and Sequence Types.

**Motivation and Context**
- Related to issue #3517

### CODE REVIEW

File: include/onnxruntime/core/session/onnxruntime_cxx_api.h [hunk 1]
```diff
@@ -74,6 +74,8 @@ ORT_DEFINE_RELEASE(RunOptions);
 ORT_DEFINE_RELEASE(Session);
 ORT_DEFINE_RELEASE(SessionOptions);
 ORT_DEFINE_RELEASE(TensorTypeAndShapeInfo);
+ORT_DEFINE_RELEASE(SequenceTypeInfo);
```
Comment (hariharans29) [inline include/onnxruntime/core/session/onnxruntime_cxx_api.h:77]: Thanks for the contribution - would you like to cover these with some tests ?

Comment (Dudeldu) [reply include/onnxruntime/core/session/onnxruntime_cxx_api.h:77]: Of course we would, but as far as we searched through it, there are no tests for the basic `TensorTypeAndShapeInfo` we could use to build upon.   
We could only further extend the `test_nontensor_types` to check the TypeInfos as well

Comment (Dudeldu) [reply include/onnxruntime/core/session/onnxruntime_cxx_api.h:77]: @hariharans29 The main functionality should now be covered.

**Comment by hariharans29** [comment]
Looks good to me, thanks

**Pull request merged**